docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+
+#### Core/Supporting
+
+- Latency (measure of time) an event takes to get through core data
+- Latency (measure of time) a command request takes to get to a device service
+- Indication of health – that events are being processed during a configurable period
+- Number of events in persistence
+- Number of readings in persistence
+- Number of validation failures (validation of device identification)
+- Number of notification transactions
+- Number of notifications handled
+- Number of failed notification transmissions
+- Number of notifications in retry status
+
+#### Application Services
+
+- Processing time for a pipeline; latency (measure of time) an event takes to get through an  application service pipeline
+- DB access times
+- How often are we failing export to be sent to db to be retried at a later time
+- What is the current store and forward queue size
+- How much data (size in KBs or MBs) of packaged sensor data is being sent to an endpoint (or volume)
+- Number of invalid messages that triggered pipeline
+- Number of events processed
+
+#### Device Services
+
+- Number of devices managed by this DS
+- Device Requests (which may be more informative than reading counts and rates)
+
+!!! Note
+    There may be additional specific metrics for each device service.  For example, the ONVIF device service may report number of times camera tampering was detected. 
+
+#### Security
+
+Security metrics may be more difficult to ascertain as they are cross service metrics.  They may have to be dealt with per service.  Also, true threat detection based on metrics may be a feature best provided by 3rd party based on particular threats and security profile needs.
+
+- Number of API requests denied due to wrong access token (Kong) per service and within a given time
+- Number of secrets accessed per service name
+- Count of any accesses and failures to the data persistence layer
+- Count of service start and restart attempts
+
+### Design Proposal
+
+#### Collect and Push Architecture
+
+Metric data will be collected and cached by each service.  At designated times (kicked off by configurable schedule), the service will collect telemetry data from the cache and push it to a designated message bus topic.
+
+#### Metrics Messaging
+
+Cached metric data, at the designated time, will be marshaled into a message and pushed to the pre-configured message bus topic.
+
+Each metric message consists of several key/value pairs:
+- a **required** name (the name of the metric) such as service-uptime
+- a **required** value which is the telemetry value collected such as 120 as the number of hours the service has been up.
+- a **required** timestamp is the time (in Epoch timestamp/milliseconds format) at which the data was collected (similar in nature to the origin of sensed data). 
+- an optional collection (array) of tags.  The tags are sets of key/value pairs of strings that provide amplifying information about the telemetry.  Tags may include:

```

Comment:
Why the added complexity? I'm not sure I see the value of tagging individual metrics?